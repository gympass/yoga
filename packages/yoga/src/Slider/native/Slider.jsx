import React from 'react';
import { View } from 'react-native';
import { bool, number, arrayOf, shape, string, oneOfType } from 'prop-types';
import { withTheme } from 'styled-components';
import MultiSlider from '@ptomasroos/react-native-multi-slider';
import { LabelView, LabelText } from './Label';
import Marker from './Marker';
import Step from './Step';

/** Sliders allow users to make selections from a range of values. Sliders
reflect a range of values along a bar, from which users may select a single
value. They are ideal for adjusting settings such as volume, brightness, or
applying image filters. */
const Slider = ({
  max,
  maxLabel,
  min,
  minLabel,
  snapped,
  values,
  tooltip,
  sliderLength,
  theme: {
    yoga: {
      components: { slider },
    },
  },
  ...props
}) => {
  const renderSnapDots = () => {
    const items = [];
    for (let i = min; i <= max; i++) {
      if (values.length > 1) {
        items.push(<Step key={i} active={values[0] < i && values[1] > i} />);
      } else {
        items.push(<Step key={i} active={values[0] > i} />);
      }
    }

    return items;
  };

  const renderTooltip = side => {
    const tooltipSide = side === 'left' ? 0 : 1;

    return tooltip.filter(({ step, visible }) => {
      if (!step && step !== 0) {
        return visible;
      }

      return visible && values[tooltipSide] === step;
    })[0];
  };

  return (
    <View style={{ width: sliderLength }}>
      {snapped && (
        <View
          style={{
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'space-between',
            bottom: -30,
          }}
        >
          {renderSnapDots()}
        </View>
      )}
      <MultiSlider
        {...props}
        values={values}
        sliderLength={sliderLength}
        trackStyle={{
          backgroundColor: slider.track.backgroundColor.inactive,
          height: 4,
          borderRadius: slider.track.border.radius,
        }}
        selectedStyle={{
          backgroundColor: slider.track.backgroundColor.active,
        }}
        isMarkersSeparated
        min={min}
        max={max}
        snapped={snapped}
        customMarkerLeft={({ pressed }) => (
          <Marker tooltip={renderTooltip('left')} pressed={pressed} />
        )}
        customMarkerRight={({ pressed }) => (
          <Marker tooltip={renderTooltip('right')} pressed={pressed} />
        )}
        touchDimensions={{
          slipDisplacement: 400,
        }}
      />
      {(minLabel || maxLabel) && (
        <LabelView>
          {minLabel && <LabelText placement="left">{minLabel}</LabelText>}
          {maxLabel && <LabelText placement="right">{maxLabel}</LabelText>}
        </LabelView>
      )}
    </View>
  );
};

Slider.propTypes = {
  /** set the max value */
  max: number,
  /** label to be displayed at max value */
  maxLabel: oneOfType([string, number]),
  /** set the min value */
  min: number,
  /** label to be displayed at min value */
  minLabel: oneOfType([string, number]),
  sliderLength: number,
  /** make slider be snap through dots */
  snapped: bool,
  /** accepts a shape with 'ribbon', 'title', 'description', 'visible', 'step' and 'alwaysShow' properties */
  tooltip: arrayOf(
    shape({
      description: string,
      title: string,
      ribbon: string,
      visible: bool,
      step: number,
      alwaysShow: bool,
    }),
  ),
  /** an array that accepts one or two numbers, this determines how many markers will be displayed */
  values: arrayOf(number),
};

Slider.defaultProps = {
  max: 10,
  maxLabel: undefined,
  min: 0,
  minLabel: undefined,
  sliderLength: 280,
  snapped: false,
  tooltip: [],
  values: [0],
};

Slider.displayName = 'Slider';

export default withTheme(Slider);
