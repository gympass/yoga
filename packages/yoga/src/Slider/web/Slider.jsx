import React from 'react';
import { bool, string, number, arrayOf, shape, oneOfType } from 'prop-types';
import styled, { withTheme } from 'styled-components';
import RCSlider from 'rc-slider/lib/Slider';
import RCRange from 'rc-slider/lib/Range';
import Marker from './Marker';

const StyledSlider = styled(RCSlider)`
  position: relative;
  width: 100%;

  ${({ marks }) => `
    padding-bottom: ${marks ? 40 : 15}px;
  `}

  * {
    box-sizing: border-box;
  }
`;

const StyledRange = styled(RCRange)`
  position: relative;
  width: 100%;

  ${({ marks }) => `
    padding-bottom: ${marks ? 40 : 15}px;
  `}

  * {
    box-sizing: border-box;
  }
`;

const emptyArr = [];
const defaultValues = [0];

/** Sliders allow users to make selections from a range of values. Sliders
reflect a range of values along a bar, from which users may select a single
value. They are ideal for adjusting settings such as volume, brightness, or
applying image filters. */
const Slider = ({
  snapped = false,
  tooltip = emptyArr,
  values = defaultValues,
  max = 10,
  maxLabel,
  min = 0,
  minLabel,
  theme: {
    yoga: {
      components: { slider },
    },
  },
  ...props
}) => {
  const styles = {
    commonTrackStyle: {
      borderRadius: slider.track.border.radius,
      height: 4,
      position: 'absolute',
      width: '100%',
    },

    trackStyle: {
      backgroundColor: slider.track.backgroundColor.active,
    },

    railStyle: {
      backgroundColor: slider.track.backgroundColor.inactive,
    },

    commonStepStyle: {
      borderRadius: slider.step.border.radius,
      marginLeft: -4,
      position: 'absolute',
      height: 10,
      top: -3,
      width: 10,
    },

    activeStepStyle: {
      backgroundColor: slider.track.backgroundColor.active,
    },

    inactiveStepStyle: {
      backgroundColor: slider.track.backgroundColor.inactive,
    },

    labelStyle: {
      position: 'absolute',
      top: 15,
    },
  };

  return values.length > 1 ? (
    <StyledRange
      {...props}
      dots={snapped}
      value={values}
      allowCross={false}
      marks={{
        [min]: {
          label: minLabel,
          style: { ...styles.labelStyle, transform: 'translateX(0%)' },
        },
        [max]: {
          label: maxLabel,
          style: { ...styles.labelStyle, transform: 'translateX(-100%)' },
        },
      }}
      trackStyle={[
        {
          ...styles.commonTrackStyle,
          ...styles.trackStyle,
        },
      ]}
      railStyle={{
        ...styles.commonTrackStyle,
        ...styles.railStyle,
      }}
      dotStyle={
        snapped
          ? {
              ...styles.commonStepStyle,
              ...styles.inactiveStepStyle,
            }
          : undefined
      }
      activeDotStyle={
        snapped
          ? {
              ...styles.commonStepStyle,
              ...styles.activeStepStyle,
            }
          : undefined
      }
      max={max}
      min={min}
      handle={({ ref, index, ...rest }) => (
        <Marker
          values={values}
          tooltip={tooltip}
          key={index}
          index={index}
          {...rest}
        />
      )}
    />
  ) : (
    <StyledSlider
      {...props}
      prefixCls="yoga"
      dots={snapped}
      value={values[0]}
      marks={{
        [min]: {
          label: minLabel,
          style: { ...styles.labelStyle, transform: 'translateX(0%)' },
        },
        [max]: {
          label: maxLabel,
          style: { ...styles.labelStyle, transform: 'translateX(-100%)' },
        },
      }}
      trackStyle={{
        ...styles.commonTrackStyle,
        ...styles.trackStyle,
      }}
      railStyle={{
        ...styles.commonTrackStyle,
        ...styles.railStyle,
      }}
      dotStyle={
        snapped
          ? {
              ...styles.commonStepStyle,
              ...styles.inactiveStepStyle,
            }
          : undefined
      }
      activeDotStyle={
        snapped
          ? {
              ...styles.commonStepStyle,
              ...styles.activeStepStyle,
            }
          : undefined
      }
      max={max}
      min={min}
      handle={({ ref, index, ...rest }) => (
        <Marker
          values={values}
          tooltip={tooltip}
          key={index}
          index={index}
          {...rest}
        />
      )}
    />
  );
};

Slider.propTypes = {
  /** set the max value */
  max: number,
  /** label to be displayed at max value */
  maxLabel: oneOfType([string, number]),
  /** set the min value */
  min: number,
  /** label to be displayed at min value */
  minLabel: oneOfType([string, number]),
  /** make slider be snap through dots */
  snapped: bool,
  /** accepts a shape with 'ribbon', 'title', 'description', 'visible' and 'step' properties */
  tooltip: arrayOf(
    shape({
      ribbon: string,
      title: string,
      description: string,
      visible: bool,
      step: number,
    }),
  ),
  /** an array that accepts one or two numbers, this determines how many markers will be displayed */
  values: arrayOf(number),
};

Slider.displayName = 'Slider';

export default withTheme(Slider);
