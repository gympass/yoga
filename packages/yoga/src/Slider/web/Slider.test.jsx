import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ThemeProvider from '../../Theme';
import Slider from './Slider';

describe('<Slider />', () => {
  describe('Snapshots', () => {
    describe('With one marker', () => {
      it('should match snapshot with default Slider', () => {
        const { container } = render(
          <ThemeProvider>
            <Slider />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with Slider with labels', () => {
        const { container } = render(
          <ThemeProvider>
            <Slider minLabel={0} maxLabel={10} />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with Slider snapped', () => {
        const { container } = render(
          <ThemeProvider>
            <Slider snapped />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });
    });
    describe('With two markers', () => {
      it('should match snapshot', () => {
        const { container } = render(
          <ThemeProvider>
            <Slider values={[3, 7]} />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with Slider snapped', () => {
        const { container } = render(
          <ThemeProvider>
            <Slider values={[3, 7]} snapped />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('Labels', () => {
    it('should render labels', () => {
      const { getByText } = render(
        <ThemeProvider>
          <Slider maxLabel="max" minLabel="min" />
        </ThemeProvider>,
      );
      expect(getByText('max').textContent).toBe('max');
      expect(getByText('min').textContent).toBe('min');
    });
  });

  describe('Tooltip', () => {
    it('should render tooltips', () => {
      const { container, getByRole, getByText } = render(
        <ThemeProvider>
          <Slider
            values={[0]}
            tooltip={[
              {
                ribbon: 'ribbon',
                title: 'title',
                description: 'description',
                visible: true,
              },
            ]}
          />
        </ThemeProvider>,
      );

      fireEvent.mouseDown(getByRole('slider'));

      expect(getByText('ribbon')).toBeTruthy();
      expect(getByText('title')).toBeTruthy();
      expect(getByText('description')).toBeTruthy();
      expect(container).toMatchSnapshot();
    });

    it('should keep the tooltip rendered if alwaysShow is truthy', () => {
      const { container, getByText } = render(
        <ThemeProvider>
          <Slider
            values={[0]}
            tooltip={[
              {
                ribbon: 'ribbon',
                title: 'title',
                description: 'description',
                visible: true,
                alwaysShow: true,
              },
            ]}
          />
        </ThemeProvider>,
      );

      expect(getByText('ribbon')).toBeTruthy();
      expect(getByText('title')).toBeTruthy();
      expect(getByText('description')).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
