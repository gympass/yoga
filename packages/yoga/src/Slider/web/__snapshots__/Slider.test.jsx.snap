// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Snapshots With one marker should match snapshot with Slider snapped 1`] = `
.c1 {
  cursor: pointer;
  height: 24px;
  margin-top: -10px;
  outline: none;
  position: absolute;
  top: 0;
  width: 24px;
  background-color: #D8385E;
  border: 6px solid #FFFFFF;
  border-radius: 9999px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
}

.c0 {
  position: relative;
  width: 100%;
  padding-bottom: 40px;
}

.c0 * {
  box-sizing: border-box;
}

<div>
  <div
    class="yoga yoga-with-marks c0"
  >
    <div
      class="yoga-rail"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 100%; background-color: rgb(215, 215, 224);"
    />
    <div
      class="yoga-track"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 0%; background-color: rgb(216, 56, 94); left: 0%;"
    />
    <div
      class="yoga-step"
    >
      <span
        class="yoga-dot yoga-dot-active"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(216, 56, 94); left: 0%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 100%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 10%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 20%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 30%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 40%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 50%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 60%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 70%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 80%;"
      />
      <span
        class="yoga-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 90%;"
      />
    </div>
    <div
      aria-disabled="false"
      aria-valuemax="10"
      aria-valuemin="0"
      aria-valuenow="0"
      class="c1 yoga-handle"
      data-dragging="false"
      role="slider"
      style="left: 0%; transform: translateX(-50%);"
      tabindex="0"
    />
    <div
      class="yoga-mark"
    />
  </div>
</div>
`;

exports[`<Slider /> Snapshots With one marker should match snapshot with Slider with labels 1`] = `
.c1 {
  cursor: pointer;
  height: 24px;
  margin-top: -10px;
  outline: none;
  position: absolute;
  top: 0;
  width: 24px;
  background-color: #D8385E;
  border: 6px solid #FFFFFF;
  border-radius: 9999px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
}

.c0 {
  position: relative;
  width: 100%;
  padding-bottom: 40px;
}

.c0 * {
  box-sizing: border-box;
}

<div>
  <div
    class="yoga yoga-with-marks c0"
  >
    <div
      class="yoga-rail"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 100%; background-color: rgb(215, 215, 224);"
    />
    <div
      class="yoga-track"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 0%; background-color: rgb(216, 56, 94); left: 0%;"
    />
    <div
      class="yoga-step"
    >
      <span
        class="yoga-dot yoga-dot-active"
        style="left: 0%;"
      />
      <span
        class="yoga-dot"
        style="left: 100%;"
      />
    </div>
    <div
      aria-disabled="false"
      aria-valuemax="10"
      aria-valuemin="0"
      aria-valuenow="0"
      class="c1 yoga-handle"
      data-dragging="false"
      role="slider"
      style="left: 0%; transform: translateX(-50%);"
      tabindex="0"
    />
    <div
      class="yoga-mark"
    >
      <span
        class="yoga-mark-text yoga-mark-text-active"
        style="transform: translateX(0%); left: 0%; position: absolute; top: 15px;"
      >
        0
      </span>
      <span
        class="yoga-mark-text"
        style="transform: translateX(-100%); left: 100%; position: absolute; top: 15px;"
      >
        10
      </span>
    </div>
  </div>
</div>
`;

exports[`<Slider /> Snapshots With one marker should match snapshot with default Slider 1`] = `
.c1 {
  cursor: pointer;
  height: 24px;
  margin-top: -10px;
  outline: none;
  position: absolute;
  top: 0;
  width: 24px;
  background-color: #D8385E;
  border: 6px solid #FFFFFF;
  border-radius: 9999px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
}

.c0 {
  position: relative;
  width: 100%;
  padding-bottom: 40px;
}

.c0 * {
  box-sizing: border-box;
}

<div>
  <div
    class="yoga yoga-with-marks c0"
  >
    <div
      class="yoga-rail"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 100%; background-color: rgb(215, 215, 224);"
    />
    <div
      class="yoga-track"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 0%; background-color: rgb(216, 56, 94); left: 0%;"
    />
    <div
      class="yoga-step"
    >
      <span
        class="yoga-dot yoga-dot-active"
        style="left: 0%;"
      />
      <span
        class="yoga-dot"
        style="left: 100%;"
      />
    </div>
    <div
      aria-disabled="false"
      aria-valuemax="10"
      aria-valuemin="0"
      aria-valuenow="0"
      class="c1 yoga-handle"
      data-dragging="false"
      role="slider"
      style="left: 0%; transform: translateX(-50%);"
      tabindex="0"
    />
    <div
      class="yoga-mark"
    />
  </div>
</div>
`;

exports[`<Slider /> Snapshots With two markers should match snapshot 1`] = `
.c1 {
  cursor: pointer;
  height: 24px;
  margin-top: -10px;
  outline: none;
  position: absolute;
  top: 0;
  width: 24px;
  background-color: #D8385E;
  border: 6px solid #FFFFFF;
  border-radius: 9999px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
}

.c0 {
  position: relative;
  width: 100%;
  padding-bottom: 40px;
}

.c0 * {
  box-sizing: border-box;
}

<div>
  <div
    class="rc-slider rc-slider-with-marks c0"
  >
    <div
      class="rc-slider-rail"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 100%; background-color: rgb(215, 215, 224);"
    />
    <div
      class="rc-slider-track rc-slider-track-1"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 40%; background-color: rgb(216, 56, 94); left: 30%;"
    />
    <div
      class="rc-slider-step"
    >
      <span
        class="rc-slider-dot"
        style="left: 0%;"
      />
      <span
        class="rc-slider-dot"
        style="left: 100%;"
      />
    </div>
    <div
      aria-disabled="false"
      aria-valuemax="10"
      aria-valuemin="0"
      aria-valuenow="3"
      class="c1 rc-slider-handle rc-slider-handle-1"
      data-dragging="false"
      role="slider"
      style="left: 30%; transform: translateX(-50%);"
      tabindex="0"
    />
    <div
      aria-disabled="false"
      aria-valuemax="10"
      aria-valuemin="0"
      aria-valuenow="7"
      class="c1 rc-slider-handle rc-slider-handle-2"
      data-dragging="false"
      role="slider"
      style="left: 70%; transform: translateX(-50%);"
      tabindex="0"
    />
    <div
      class="rc-slider-mark"
    />
  </div>
</div>
`;

exports[`<Slider /> Snapshots With two markers should match snapshot with Slider snapped 1`] = `
.c1 {
  cursor: pointer;
  height: 24px;
  margin-top: -10px;
  outline: none;
  position: absolute;
  top: 0;
  width: 24px;
  background-color: #D8385E;
  border: 6px solid #FFFFFF;
  border-radius: 9999px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
}

.c0 {
  position: relative;
  width: 100%;
  padding-bottom: 40px;
}

.c0 * {
  box-sizing: border-box;
}

<div>
  <div
    class="rc-slider rc-slider-with-marks c0"
  >
    <div
      class="rc-slider-rail"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 100%; background-color: rgb(215, 215, 224);"
    />
    <div
      class="rc-slider-track rc-slider-track-1"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 40%; background-color: rgb(216, 56, 94); left: 30%;"
    />
    <div
      class="rc-slider-step"
    >
      <span
        class="rc-slider-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 0%;"
      />
      <span
        class="rc-slider-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 100%;"
      />
      <span
        class="rc-slider-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 10%;"
      />
      <span
        class="rc-slider-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 20%;"
      />
      <span
        class="rc-slider-dot rc-slider-dot-active"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(216, 56, 94); left: 30%;"
      />
      <span
        class="rc-slider-dot rc-slider-dot-active"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(216, 56, 94); left: 40%;"
      />
      <span
        class="rc-slider-dot rc-slider-dot-active"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(216, 56, 94); left: 50%;"
      />
      <span
        class="rc-slider-dot rc-slider-dot-active"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(216, 56, 94); left: 60%;"
      />
      <span
        class="rc-slider-dot rc-slider-dot-active"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(216, 56, 94); left: 70%;"
      />
      <span
        class="rc-slider-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 80%;"
      />
      <span
        class="rc-slider-dot"
        style="border-radius: 9999px; margin-left: -4px; position: absolute; height: 10px; top: -3px; width: 10px; background-color: rgb(215, 215, 224); left: 90%;"
      />
    </div>
    <div
      aria-disabled="false"
      aria-valuemax="10"
      aria-valuemin="0"
      aria-valuenow="3"
      class="c1 rc-slider-handle rc-slider-handle-1"
      data-dragging="false"
      role="slider"
      style="left: 30%; transform: translateX(-50%);"
      tabindex="0"
    />
    <div
      aria-disabled="false"
      aria-valuemax="10"
      aria-valuemin="0"
      aria-valuenow="7"
      class="c1 rc-slider-handle rc-slider-handle-2"
      data-dragging="false"
      role="slider"
      style="left: 70%; transform: translateX(-50%);"
      tabindex="0"
    />
    <div
      class="rc-slider-mark"
    />
  </div>
</div>
`;

exports[`<Slider /> Tooltip should render tooltips 1`] = `
.c2 {
  width: 95px;
  position: absolute;
  bottom: 30px;
  left: 6px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  background-color: #D8385E;
  border-radius: 8px;
  box-shadow: 0 5px 5px -3px rgba(152,152,166,0.2),0 8px 10px 1px rgba(152,152,166,0.14),0 3px 14px 2px rgba(152,152,166,0.12);
}

.c4 {
  position: relative;
  z-index: 3;
  background-color: #F5F5FA;
  border-radius: 8px;
  padding: 4px 8px 4px 8px;
}

.c7 {
  position: absolute;
  bottom: -5px;
  left: 50%;
  z-index: 2;
  width: 16px;
  height: 16px;
  border-radius: 3px;
  -webkit-transform: rotate(45deg) translateX(-50%);
  -ms-transform: rotate(45deg) translateX(-50%);
  transform: rotate(45deg) translateX(-50%);
  -webkit-transform-origin: left;
  -ms-transform-origin: left;
  transform-origin: left;
  background-color: #F5F5FA;
  box-shadow: 0 5px 5px -3px rgba(152,152,166,0.2),0 8px 10px 1px rgba(152,152,166,0.14),0 3px 14px 2px rgba(152,152,166,0.12);
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 700;
  position: relative;
  text-align: center;
  display: block;
  padding: 4px 8px 4px 8px;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
  color: #FFFFFF;
  font-size: 12px;
}

.c5 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  display: block;
  text-align: center;
  color: #6B6B78;
  font-weight: 400;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  display: block;
  text-align: center;
  color: #6B6B78;
  font-weight: 400;
}

.c1 {
  cursor: pointer;
  height: 24px;
  margin-top: -10px;
  outline: none;
  position: absolute;
  top: 0;
  width: 24px;
  background-color: #D8385E;
  border: 6px solid #FFFFFF;
  border-radius: 9999px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12), 0 0 0 8px rgba(216,56,94,0.2);
}

.c0 {
  position: relative;
  width: 100%;
  padding-bottom: 40px;
}

.c0 * {
  box-sizing: border-box;
}

<div>
  <div
    class="yoga yoga-with-marks c0"
  >
    <div
      class="yoga-rail"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 100%; background-color: rgb(215, 215, 224);"
    />
    <div
      class="yoga-track"
      style="border-radius: 9999px; height: 4px; position: absolute; width: 0%; background-color: rgb(216, 56, 94); left: 0%;"
    />
    <div
      class="yoga-step"
    >
      <span
        class="yoga-dot yoga-dot-active"
        style="left: 0%;"
      />
      <span
        class="yoga-dot"
        style="left: 100%;"
      />
    </div>
    <div
      aria-disabled="false"
      aria-valuemax="10"
      aria-valuemin="0"
      aria-valuenow="0"
      class="c1 yoga-handle"
      data-dragging="true"
      role="slider"
      style="left: 0%; transform: translateX(-50%);"
      tabindex="0"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          ribbon
        </span>
        <div
          class="c4"
        >
          <span
            class="c5"
          >
            title
          </span>
          <span
            class="c6"
          >
            description
          </span>
        </div>
        <div
          class="c7"
        />
      </div>
    </div>
    <div
      class="yoga-mark"
    />
  </div>
</div>
`;
